libraries/cms/application/cms.php

```diff
@@ -114,103 +114,15 @@ public function __construct(JInput $input = null, Registry $config = null, JAppl
 		}
 
 		// Enable sessions by default.
-		if (is_null($this->config->get('session')))
+		if (is_null($this->getContainer()->get('config')->get('session')))
```

Comment:
I made a revert commit with the Registry setting on the application PR after rereading your comments. It should be better now.
But I mean if my approach is completely wrong then let's close the two PR's, looks like I got something wrong.